2-mandatory/3-financial-times.js

```diff
@@ -14,24 +27,52 @@ function potentialHeadlines(allArticleTitles) {
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+  // TODO
+  let fewestWordArticle = allArticleTitles[0];
+  for (let i = 0; i < allArticleTitles.length; i++) {
+    if (allArticleTitles[i].length < fewestWordArticle.length) {
+      fewestWordArticle = allArticleTitles[i];
+    }
+  }
+  return fewestWordArticle;
 }
 
+
+
+
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
-    Implement the function below to return a new array containing all the headlines which contain a number.
+    Implement the function below to return a new array containing all the headlines which contain 
+    a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+  // TODO
+  let newArray = [];
+  for (let i = 0; i < allArticleTitles.length; i++) {
+    if (/[0-9]/g.test(allArticleTitles[i])) {
```

Comment:
using regex here makes sense, well done. You can also use `\d` for checking for digits, I find this site really useful for trying out different regex patterns https://regex101.com/

Reply:
thank you